### PR TITLE
fix: Align the 'Editable' checkbox on first linking data to a table widget

### LINKED ISSUE #15384: [Bug]: Editable checkbox is not aligned with rest of the column checkboxes in Table widget
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description

When we D&D a Table widget to the canvas and bind it with a query, the checkbox next to the `Editable` property does not align with the checkboxes of other columns.

https://www.loom.com/share/d467f54dd72048e5b7f53055a5d7cb70

### Steps To Reproduce

1. D&D a Table widget.
2. Bind it with a query.

**Expected:** The checkboxes for the editable property should all be properly aligned.
**Actual:** Initially on binding the Table widget with a query for the first time, the checkbox next to the `Editable` keyword is misaligned. Once we refresh the page, the checkbox gets aligned automatically. 

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by eloyrobillard**
## Description

Before the change, linking data to a table widget for the first time led to the 'Editable' checkbox being farther to the right :

first time linking

than after reloading the page:

after reloading the page

After the changes, the button always appears as in the bottom image. The issue was due to special padding being applied to scrollable lists of columns (lists taller than the maximum size of the element). The padding was only applied after a component update, but not at mounting time. 

Note that the checkbox is never aligned with the checkboxes below it for me (I use Chrome on macOS 14). This is slightly different than for the person who reported the original issue.

#### PR fixes following issue(s)
Fixes #15384 

#### Type of change
Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?

- [x] Manual

To test this out:

- [ ] Add a new table widget to your app
- [ ] Link a data source under `Table data` on the right (it must have enough columns that the `Columns` list is scrollable)
- [ ] Take notice of the horizontal position of the `Editable` checkbox relative to the columns below it
- [ ] Reload the page
- [ ] The `Editable` checkbox should be in the exact same spot 

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved scrollable lists handling in the `PrimaryColumnsControlV2` component.

### CODE REVIEW

File: app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx [hunk 1]
```diff
@@ -114,6 +114,23 @@ class PrimaryColumnsControlV2 extends BaseControl<ControlProps, State> {
     };
   }
 
+  componentDidMount(): void {
+    const listElement = document.querySelector(`.${LIST_CLASSNAME}`);
```
Comment (sbalaji1192) [inline app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:118]: Can you extract this into a function, call here and in componentDidUpdate functions?

Comment (eloyrobillard) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx]: done

Comment (sbalaji1192) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx]: Thanks. Let me trigger the tests.

**Comment by sbalaji1192** [comment]
@eloyrobillard Can you merge the latest release to this branch?

**Comment by eloyrobillard** [comment]
merged

**Comment by eloyrobillard** [comment]
@sbalaji1192 merged

**Review by sbalaji1192 — APPROVED** [review]

**Comment by sbalaji1192** [comment]
@eloyrobillard I have merged your changes. Feel free to take a look at any other issues.

**Pull request merged**